.circleci/config.yml

```diff
@@ -16,15 +16,19 @@ jobs:
 
       - run: npm install
       - run: npm run linter:check
-      - run: npm run release:test
-      - run: node build.js '0.0.1'
+      - run: npm run build:package
+      - run: npm run build:storybook
+      # check npm token exists
+      - run: echo "//registry.npmjs.org/:_authToken=$npm_TOKEN" > ~/repo/.npmrc
```

Comment:
found that npm actually can read from the environment variables to interpolate this string by himself.
so better to do it as `echo '//registry.npmjs.org/:_authToken=${npm_TOKEN}' > ~/repo/.npmrc` (with single quotes)

Reply:
Except that circle will inject the token and it does in fact insert the token and not a variable. I was double checking this was true with the cat command (because I also was skeptical). If we expose it in brackets will that cause any sort of leak?

Reply:
wut? I can't understand you.

- in build you have environment variables
- they can be expanded by shell or by npm
- with shell you will persist the token - and this is bad as it can be cached, or someone could have access to the fs
- so if npm can expand by himself this variable, why ever save it expanded?
- the curly brackets is how actually env variables should be written if they are part of the sting (code-style)

Reply:
I’ll come to you to discuss but I think I get what you’re saying. I got lost in the last bit my bad.